src/test/java/bio/terra/app/controller/SnapshotsApiControllerTest.java

```diff
@@ -261,6 +279,181 @@ void unlockSnapshot() throws Exception {
     verify(snapshotService).manualExclusiveUnlock(TEST_USER, SNAPSHOT_ID, unlockRequest);
   }
 
+  @Test
```

Comment:
I noticed that `SnapshotsApiControllerTest.java` is about ~60% covered. Are there other tests that could be added to increase confidence to ultimately run tests on BEEs?

Reply:
I'm curious, can you point me to where you are seeing the coverage figure?  When I click into the SonarCloud analysis I can't see what the coverage would be if I merged, but I can see that the [existing coverage](https://sonarcloud.io/code?id=DataBiosphere_jade-data-repo&selected=DataBiosphere_jade-data-repo%3Asrc%2Fmain%2Fjava%2Fbio%2Fterra%2Fapp%2Fcontroller%2FSnapshotsApiController.java) in `develop` branch is much lower at 30%.

TDR's test base has historically been integration-heavy which gives us test confidence at a cost (integration and connected tests take a long time to run on PRs).  I don't think that shifting the balance to more unit test coverage is itself a requirement to run tests on BEEs, but it's something we want to do for other reasons.

I'll file a ticket to evaluate further test coverage on this controller, with the understanding that more will follow.

Reply:
Follow-on ticket: https://broadworkbench.atlassian.net/browse/DCJ-267

Reply:
> I'm curious, can you point me to where you are seeing the coverage figure?

There is a coverage report on the PR page in sonar (https://sonarcloud.io/summary/new_code?id=DataBiosphere_jade-data-repo&pullRequest=1647), although as I load it right now I don't see data but I think that's because the build step hasn't yet pushed the coverage data up. You can see coverage for other branches, like this one https://sonarcloud.io/summary/new_code?id=DataBiosphere_jade-data-repo&pullRequest=1668

Reply:
I ran the coverage for the file in IntelliJ and extracted the covered lines -- I didn't spend too much time on it so it's definitely possible the figure is incorrect. The main thrust of my concern was less to do with the specific percentage and more to do with are there test cases that would be valuable to have as we move towards running things on BEEs.